go/database/sql/connection.go

```diff
@@ -89,6 +102,15 @@ func (s *sqlCommenterConn) Raw() driver.Conn {
 	return s.Conn
 }
 
+func (s *sqlCommenterConnWithTx) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, error) {
+	begintxer, ok := s.Conn.(driver.ConnBeginTx)
+	if !ok {
+		return nil, driver.ErrSkip
+	}
```

Comment:
Do we really need an extra driver config ?

Will this condition not take care of any driver not implementing `ConnBeginTx' ?

Reply:
What i meant is do we need to create a new struct `sqlCommenterConnWithTx` ?

Can we just not implement `BeginTx` with `sqlCommenterConn` ?
```
func (s *sqlCommenterConn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, error) {...}
```

Reply:
I understand that :).
It's mainly your call. 
Doing so makes the code simpler but it may change the expected behavior for drivers that don't implement `BeginTx` as the code in go library has a special handling for drivers that don't implement that interface.